src/lib/remote-place-cache.ts

```diff
@@ -9,18 +12,36 @@ type RemotePlaceDatastore = {
   [key: string]: RemotePlacesByType;
 };
 
+export type RemotePlace = {
+  id: string;
+  name: IPropertyValue;
+  lineage: string[];
+  ambiguities?: RemotePlace[];
+
+  // sadly, sometimes invalid or uncreated objects "pretend" to be remote
+  // should reconsider this naming
+  type: 'remote' | 'local' | 'invalid';
+};
+
 export default class RemotePlaceCache {
   private static cache: RemotePlaceDatastore = {};
 
-  public static async getPlacesWithType(chtApi: ChtApi, placeType: string)
+  public static async getPlacesWithType(chtApi: ChtApi, contactType: ContactType, hierarchyLevel: HierarchyConstraint)
     : Promise<RemotePlace[]> {
-    const domainStore = await RemotePlaceCache.getDomainStore(chtApi, placeType);
+    const domainStore = await RemotePlaceCache.getDomainStore(chtApi, contactType, hierarchyLevel);
     return domainStore;
   }
 
-  public static async add(place: Place, chtApi: ChtApi): Promise<void> {
-    const domainStore = await RemotePlaceCache.getDomainStore(chtApi, place.type.name);
-    domainStore.push(place.asRemotePlace());
+  public static add(place: Place, chtApi: ChtApi): void {
```

Comment:
Since we can figure out if a `local` place is created on a remote instance, do we still need this?

Reply:
The reason we need `add` is so we don't have to `refresh` the full cache (heavy requests to instance) after creating a place through the management tool. We just shim the "local one" into the cache cache and it pretends to be remote (which it is, if everything went well).

Reply:
Yes i understand that, the reason i bring it up is the comments here

> // sadly, sometimes invalid or uncreated objects "pretend" to be remote
>   // should reconsider this naming
>   type: 'remote' | 'local' | 'invalid';

I noticed we can convert a Place to a Remote place even when it's not created on the server. I think it'd make sense to treat remote places strictly as places we fetched from the server then we could rid of the `type` field. During search, the remote places can be joined with places that have `isCreated` set to true (maybe under an interface defining the fields needed?). Does that make sense?

Reply:
> During search, the remote places can be joined with places that have isCreated set to true (maybe under an interface defining the fields needed?). Does that make sense?

I've tried this, and it's a bit complex. Mostly because you need the configuration object to get data from the Place into a similar structure that comes in RemotePlace.  I spent like an entire day on it and I think you're right that what we have here is still unsatisfying. 

I suspect `Place` should be called something like `StagedPlace` or something... since it represents a row in the table which is going to be created.  I think `RemotePlace` initially represented only a place from CHT. I think it should be a class instead of a type.   And then I think we need a third type which they both implement like `SearchablePlace` (or something alike)... 

I think this is probably a lot of code churn and this PR is already large + is the base of #20.  Would you mind if we address it in a follow-up ticket?

Reply:
@freddieptf Thoughts?

Reply:
Yes i think that can be addressed in another PR